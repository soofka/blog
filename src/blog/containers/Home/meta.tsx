import { inject, observer } from 'mobx-react';
import * as React from 'react';

import { getLabel } from 'blog/common/helpers';
import { LanguageStoreInterface } from 'blog/store/language';

import { Meta } from 'blog/components/meta';

interface HomeMetaPropsInterface {
  languageStore?: LanguageStoreInterface;
}

export const HomeMetaComponent = (props: HomeMetaPropsInterface) => {
  const { languageStore: { getLanguage } } = props;
  return <Meta
    title={getLabel('page_title', {}, getLanguage())}
    description={getLabel('page_description', { description: '' }, getLanguage())}
    keywords={getLabel('page_keywords', { keywords: '' }, getLanguage())}
  />;
};

export const HomeMeta = inject('languageStore')(observer(HomeMetaComponent));
