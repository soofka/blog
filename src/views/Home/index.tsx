import * as React from 'react';

import Entries from 'views/Entries';

export class Home extends Entries {}

export default Home;
