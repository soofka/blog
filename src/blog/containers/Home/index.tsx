import { reaction } from 'mobx';
import { inject } from 'mobx-react';
import * as React from 'react';
import { RouteComponentProps } from 'react-router-dom';

import { LanguageStoreInterface } from 'blog/store/language';

import { EntriesList } from 'blog/components/EntriesList';
import { ErrorBox } from 'blog/components/ErrorBox';
import { Label } from 'blog/components/Label';
import { LoadingCover } from 'blog/components/LoadingCover';
import { HomeMeta } from './meta';

import {
  getEntries,
  RequestSuccessResponseInterface,
  validateRequestSuccessResponse,
} from 'blog/common/helpers';

import { EntryInterface } from 'common/types';

export interface HomePropsInterface extends RouteComponentProps<any> {
  languageStore?: LanguageStoreInterface;
}

interface HomeStateInterface {
  entries: EntryInterface[];
  loading: boolean;
}

export class HomeContainer extends React.Component<HomePropsInterface, HomeStateInterface> {
  constructor(props: HomePropsInterface) {
    super(props);
    this.state = { entries: undefined, loading: false };

    reaction(
      () => this.props.languageStore.language,
      () => {
        this.getEntries();
      },
    );
  }

  componentDidMount() {
    if (!this.state.entries) {
      this.getEntries();
    }
  }

  getEntries() {
    this.setState({ loading: true });
    getEntries(this.props.languageStore.getLanguage())
      .then((response: RequestSuccessResponseInterface) => {
        const entries = validateRequestSuccessResponse(response, true)
          ? response.data
          : undefined;
        this.setState({ entries, loading: false });
      });
  }

  render() {
    const { entries, loading } = this.state;

    if (loading) {
      return (
        <LoadingCover />
      );
    }

    if (!entries) {
      return (
        <ErrorBox
          message={<Label name="errors__no_entries_in_db" />}
        />
      );
    }

    return (
      <div>
        <HomeMeta />
        <EntriesList entries={filterPrivateEntries(entries)} />
      </div>
    );
  }
}

const filterPrivateEntries = (entries: EntryInterface[]): EntryInterface[] =>
  entries.filter((entry: EntryInterface) => entry.public);

export const Home = inject('languageStore')(HomeContainer);
