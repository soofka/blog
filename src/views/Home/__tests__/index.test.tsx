import * as React from 'react';
import { shallow } from 'enzyme';

import Home from 'views/Home';

describe('<Home/>', () => {

  it('renders correctly', () => {
    expect(shallow(<Home/>)).toMatchSnapshot();
  });

});
