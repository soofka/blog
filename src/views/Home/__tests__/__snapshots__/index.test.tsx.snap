exports[`<Home/> renders correctly 1`] = `<Component />`;
